.github/workflows/ci.yml

```diff
@@ -79,11 +80,28 @@ jobs:
     if: github.event_name == 'pull_request'
     steps:
       - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0 # checkout all commits to be able to determine merge base
+      - uses: actions/setup-java@v3
+        with:
+          distribution: 'zulu'
+          java-version: 17
+          cache: 'maven'
+      - name: Fetch base ref to find merge-base for GIB
+        run: .github/bin/git-fetch-base-ref.sh
       - name: Check Commits
         uses: trinodb/github-actions/block-commits@c2991972560c5219d9ae5fb68c0c9d687ffcdd10
         with:
           action-merge: fail
           action-fixup: none
+      - name: Check if all commits of this PR compile
+        run: |
+          git config user.name "$(git --no-pager log --format=format:'%an' -n 1)"
+          git config user.email "$(git --no-pager log --format=format:'%ae' -n 1)"
+          git rebase --exec "git status && $MAVEN test-compile $MAVEN_COMPILE_COMMITS" refs/remotes/origin/${{ github.event.pull_request.base.ref }}
```

Comment:
Use `package`/`install` instead of `test-compile`. That's what people need to run during bisect and possibly some plugins are bound to those phases.

Reply:
Also it might be more useful to `$MAVEN || git status` so that `git status` is the last thing that's present in the log for a failing commit. Currently maven output would be the last thing (which makes us scroll up to find the commit).

Reply:
Should a line also be printed at the end that the commit hashes may not exist in repo (since we rebased?) and to only use the commit message for identifying?

Reply:
Wouldn't `$MAVEN || git status` prevent the whole job from failing? That should be `$MAVEN || (git status && false)`

Reply:
> Use `package`/`install` instead of `test-compile`.

I'm fine with `package`, but is it worth the additional time?
we still disable packaging-related checks, so I don't realistically see a way `package` could fail, but it definitely adds CI time

Reply:
> Currently maven output would be the last thing (which makes us scroll up to find the commit).

Not exactly true:

```
$ git rebase --exec 'echo boo; false' HEAD^^^
Executing: echo boo; false
boo
warning: execution failed: echo boo; false
You can fix the problem, and then run

  git rebase --continue
```

but indeed it doesn't say at which commit we stopped.
(OTOH it explicitly says _why_ it stopped; `git status` will make this information harder to find... fine by me)